packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -641,7 +641,12 @@ class _AppBarState extends State<AppBar> {
         fit: StackFit.passthrough,
         children: <Widget>[
           widget.flexibleSpace,
-          appBar,
+          // Creates a material widget to prevent the flexibleSpace from swallow
+          // the ink splash effect.
+          Material(
```

Comment:
Shouldn't we be using the Ink widget here?

https://api.flutter.dev/flutter/material/MaterialType-class.html

Reply:
Ah nice

Reply:
I took a look at the ink widget, and couldn't figure out how to apply it here. If i understand correctly, the Ink widget just make the splash go through the decoration of the Ink. but there is a stack on top of the material widget

Reply:
You're right. I thought Ink would draw the splash on top of the Material's descendants, but that's not right. Your original approach was the right one after all, sorry about the detour.